webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala

```diff
@@ -188,9 +188,9 @@ class HistorizationServiceImpl(
          }
       }.toSeq.map { case (t,fat,d) => (d, fat.toActiveTechnique, t) }
 
-      val stringDirectiveIds = directives.keySet.map( _.value)
+      val stringDirectiveIds = directives.keysIterator.toSet.map( _.value)
```

Comment:
wow, this fails 
[ERROR] /root/dev/rudder/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala:191: error: missing parameter type for expanded function ((x$4: <error>) => x$4.value)
[ERROR]       val stringDirectiveIds = directives.keysIterator.toSet.map( _.value)
[ERROR]                                                                   ^

i have no idea why ...

Reply:
doing it in 2 steps fixes the issues
      val directiveSet = directives.keysIterator.toSet
      val stringDirectiveIds = directiveSet.map( _.value)